server/src/main/java/tw/waterball/survive/io/server/worldobject/role/Player.java

```diff
@@ -1,89 +1,97 @@
 package tw.waterball.survive.io.server.worldobject.role;
 
-import tw.waterball.survive.io.server.Client;
-import tw.waterball.survive.io.server.ClientSocketAdapter;
-import tw.waterball.survive.io.server.worldobject.WorldObject;
-import tw.waterball.survive.io.server.worldobject.weapon.Weapon;
 
-public class Player extends Role implements Client {
-    private final Client client = new ClientSocketAdapter();
+import tw.waterball.survive.io.api.api.ClientApi;
+import tw.waterball.survive.io.api.dto.RoleDTO;
+import tw.waterball.survive.io.api.dto.TerrainDTO;
+import tw.waterball.survive.io.api.dto.event.*;
+
+import java.util.List;
+
+public class Player extends Role implements ClientApi {
+    private final ClientApi clientApi;
+
+    public Player(ClientApi clientApi) {
+        this.clientApi = clientApi;
+    }
 
     public void reload() {
 
     }
 
     @Override
-    public void onLoginSuccessfully() {
-        client.onLoginSuccessfully();
+    public void onLoginSuccessfully(RoleDTO roleDTO) {
```

Comment:
`loginPlayerDTO`, this is why you have to use ctrl+F6

Reply:
Ok thanks